### PR TITLE
Fixed parseList error on parsing empty list items

### PR DESCRIPTION
**Pull request by AsishRaju**
### Fix

- Added a check when checking for `paragraph.type` which will return an empty notion block as wanted

### Debugging Process

- Since the error screenshot on #2 was indicating the undefined value for the variable `type`, so moving a little upon context found this condition 👇🏻 in `build\src\parser\internal.js` 

 > image

- Found that there was an assignment of the `paragraph` variable for which type was checked and this is where it failed.

 > image

- On console logging, the items that are being passed on to `parseList()`  found that listItems that have text attached to it (first object in the screenshot) have a `children` array and empty listItem (second object in the screenshot) doesn't 👇🏻
[Here](https://jsonformatter.org/1ff5dc) is the complete JSON .

 > image

<hr>

> **PS**
> I don't think it's a top-level fix for the issue; something to check with `remark-gfm` lib which is used to do the actual parsing of markdown

### CODE REVIEW

File: src/notion/blocks.ts [hunk 1]
```diff
@@ -79,3 +80,16 @@ export function toDo(checked: boolean, text: RichText[]): ToDoBlock {
...
+    object: 'block',
+    type: 'image',
+    image: {
+      type: 'external',
+      external: {
+        url: url
+      }
+    }
+  } as ImageBlock;
+}
```
Comment (juliarosenson) [inline src/notion/blocks.ts:95]: Can you add an end of file newline here?

File: src/parser/internal.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable prettier/prettier */
 import * as md from '../markdown';
```
Comment (juliarosenson) [inline src/parser/internal.ts:2]: I think we'd like to keep prettier enabled and maybe revisit the config if the settings aren't readable

File: src/parser/internal.ts [hunk 3]
```diff
@@ -105,8 +106,27 @@ function parseNode(node: md.FlowContent): notion.Block[] {
...
+          }
+        })
+        const imageBlocks = imageElement.map((element) => {
+          if (element.type === 'image') {
+            return notion.image(element.url)
+          } else {
+            return notion.image('')
+          }
+        })
+        return [...imageBlocks, parseParagraph(node)]
```
Comment (juliarosenson) [inline src/parser/internal.ts:124]: What do you think about pulling this out into a separate case and the logic into its own function for parsing images so we can maintain the simple style here?

File: test/parser.spec.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import * as md from '../src/markdown';
-import {text} from '../src/markdown';
+import { text } from '../src/markdown';
```
Comment (juliarosenson) [inline test/parser.spec.ts:2]: Can we revert these style changes for consistency across the repo?

**Review by juliarosenson — COMMENTED** [review]
Thanks for this! Would like to focus this PR on adding support for images and merge #4 to fix the parsing list error. Could you add a test for images as well as an update to the README about this new feature?

**Comment by AsishRaju** [comment]
Hey @juliarosenson, thanks for your valuable time and feedback on this pr, really appreciate it.
Sorry for polluting this pr with commits not relating to this pr, nevertheless will make a new clean pr for adding image support after working on your feedback.

Closing this pr for now. Thanks 🙂

**Pull request closed without merging**